tests/test_set_lossfunction.py

```diff
@@ -0,0 +1,184 @@
+import tempfile
+import shutil
+import unittest
+import pytest
+import logging
+import warnings
+from torch import nn
+from deeprankcore.trainer import Trainer
+from deeprankcore.dataset import GraphDataset
+from deeprankcore.neuralnets.gnn.naive_gnn import NaiveNetwork
+from deeprankcore.domain import targetstorage as targets, losstypes as losses
+
+_log = logging.getLogger(__name__)
+
+
+model_path = './tests/test.pth.tar'
+hdf5_path = 'tests/data/hdf5/test.hdf5'
+
+def base_test(trainer: Trainer, lossfunction = None, override = False):
+
+    if lossfunction:
+        trainer.set_lossfunction(lossfunction = lossfunction, override_invalid=override)
+
+    # check correct passing to/picking up from pretrained model 
+    with warnings.catch_warnings(record=UserWarning):
+        trainer.train(nepoch=2, save_best_model=None)
+        trainer.save_model(model_path)
+
+        trainer_pretrained = Trainer(
+            neuralnet = NaiveNetwork,
+            dataset_test=trainer.dataset_train,
+            pretrained_model=model_path)
+        
+        return trainer_pretrained
+
+
+class TestTrainer(unittest.TestCase):
+    @classmethod
+    def setUpClass(class_):
+        class_.work_directory = tempfile.mkdtemp()
```

Comment:
What is this directory used for?

Reply:
I think it's not used here, so it can be deleted. Also I would call the class differently, referring to the loss function rather than to the Trainer only.

Reply:
yeah, I just copy pasted from the test_trainer and didn't pay close attention to unused stuff.